mobile_index/src/indexer.rs

```diff
@@ -0,0 +1,102 @@
+use crate::{Result, Settings};
+use chrono::{DateTime, Duration, TimeZone, Utc};
+use db_store::MetaValue;
+use file_store::{FileStore, FileType, Stream};
+use futures::{stream, StreamExt, TryStreamExt};
+use helium_proto::{Message, SubnetworkReward, SubnetworkRewards};
+use sqlx::{Pool, Postgres};
+use tokio::time;
+
+pub const DEFAULT_START_REWARD_BLOCK: i64 = 1477650;
+/// default minutes to delay lookup from now
+pub const DEFAULT_LOOKUP_DELAY: i64 = 30;
+
+pub struct Indexer {
+    pool: Pool<Postgres>,
+    interval: time::Duration,
+    verifier_store: FileStore,
+}
+
+impl Indexer {
+    pub async fn new(settings: &Settings) -> Result<Self> {
+        let pool = settings.database.connect(10).await?;
+        Ok(Self {
+            interval: settings.interval(),
+            verifier_store: FileStore::from_settings(&settings.verifier).await?,
+            pool,
+        })
+    }
+
+    pub async fn run(&mut self, shutdown: triggered::Listener) -> Result {
+        tracing::info!("starting mobile indexer");
+
+        let mut interval_timer = tokio::time::interval(self.interval);
+
+        loop {
+            if shutdown.is_triggered() {
+                tracing::info!("stopping mobile indexer");
+                return Ok(());
+            }
+
+            tokio::select! {
+                _ = shutdown.clone() => (),
+                _ = interval_timer.tick() => {
+                    self.process_clock_tick().await?
+                }
+            }
+        }
+    }
+
+    async fn process_clock_tick(&mut self) -> Result {
+        let last_known_timestamp = MetaValue::<DateTime<Utc>>::fetch_or_insert_with(
+            &self.pool,
+            "last_verifier_timestamp",
+            || Utc.timestamp(0, 0),
+        )
+        .await?
+        .value()
+        .to_owned();
+
+        let time_range = last_known_timestamp..Utc::now();
```

Comment:
need to check-point somehow that this `now` was the cutoff for the collection of files processed from the bucket